### PR TITLE
Update transeq.f90

### PR DESCRIPTION
**Pull request by mathrack**
Save CPU time in `momentum_gravity` if the richardson number is zero or gravity is aligned with the axis.

Please note that I have not updated the indentation to avoid a large modification. Please let me know if updating the indentation is needed.

Cheers

### CODE REVIEW

**Review by slaizet — APPROVED** [review]
fine by me!

File: src/transeq.f90 [hunk 1]
```diff
@@ -737,7 +737,11 @@ subroutine momentum_gravity(dux1, duy1, duz1, peculiar_density1, richardson)
     integer :: iend, jend, kend
     integer :: i, j, k
 
+    ! Return directly if richardson is zero
+    if (abs(richardson) < epsilon(richardson)) return
```
Comment (pbartholomew08) [inline src/transeq.f90:741]: Should be `tiny` rather than `epsilon` to be "the smallest possible non-zero number"

**Review by pbartholomew08 — CHANGES_REQUESTED** [review]
Should use `tiny` rather than `epsilon` to be "the smallest possible non-zero number"

**Comment by mathrack** [comment]
Ready for "Squash and merge"

**Review by pbartholomew08 — APPROVED** [review]

**Pull request merged**